### ISSUE TITLE
Nullable warnings for default(T) but not default

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#38339]
**Version Used**: 
3.3.0-beta3-19415-01+f5ba9f2c61a2fe853dc4913888d40df221539147

**Steps to Reproduce**:
```C#
#nullable enable
using System.Diagnostics.CodeAnalysis;

class Program
{
    static void Main() { }
}

class C<T>
{
    [AllowNull]
    internal T _f1 = default(T);
    [AllowNull]
    internal T _f2 = default;
}
```

**Expected Behavior**:
No warnings.

**Actual Behavior**:
Warning on the `default(T)`.
```
warning CS8653: A default expression introduces a null value when 'T' is a non-nullable reference type.
```

cc: @jcouv, @cston

**Comment by Bosch-Eli-Black** [comment]
Not sure if this is related, but the below code triggers the warning "CS8653 A default expression introduces a null value when 'T' is a non-nullable reference type."

```
private T Foo<T>() {
   return default;
}
```

But this doesn't trigger any warnings at all:

```
private T Foo<T>(T defaultValue = default) {
   return defaultValue;
}
```

**Comment by cston** [comment]
@Bosch-Eli-Black, thanks for reporting the missing warning for the optional parameter value. I've logged #40818 for that particular issue.

**Comment by cston** [comment]
The original issue is fixed in 16.5.
